packages/core/src/controllers/NetworkController.ts

```diff
@@ -94,6 +95,9 @@ export const NetworkController = {
   async switchActiveNetwork(network: NetworkControllerState['caipNetwork']) {
     await this._getClient().switchCaipNetwork(network)
     state.caipNetwork = network
+    if (ApiController && CoreHelperUtil.checkCaipNetwork(network, 'solana')) {
+      await ApiController.reFetchWallets()
+    }
```

Comment:
curious: why is a reset required here for solana specifically?

Reply:
Without refetching here, when disconnected user change networks, it shows wrong wallets count in the modal all wallets.

Reply:
@gooxpf In this case, shouldn't it be refetching when switch from Solana -> Ethereum? Thise code condition, it'll only refetch when switch from X -> Solana right?

Reply:
Yes, moved it out to solana listener when chain is changed, also fixed w3m-connect-view, added subscription to  wallets count